src/index.js

```diff
@@ -0,0 +1,154 @@
+/**
+*
+* @licstart  The following is the entire license notice for the JavaScript code in this file.
+*
+* Copyright 2023 University Of Helsinki (The National Library Of Finland)
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+
+import {readFileSync} from 'fs';
+import {v4 as uuid} from 'uuid';
+
+import passport from 'passport';
+import {BasicStrategy} from 'passport-http';
+import {Strategy as BearerStrategy} from 'passport-http-bearer';
+
+import {createLogger} from '@natlibfi/melinda-backend-commons';
+import {clone} from '@natlibfi/melinda-commons';
+import {KeycloakStrategy} from '@natlibfi/passport-keycloak';
+
+/**
+ * Derived from passport-melinda-crowd-js src/index.js (https://github.com/NatLibFi/passport-melinda-crowd-js/blob/master/src/index.js)
+ *   - Copyright (C) 2018-2020 University of Helsinki (The National Library of Finland)
+ */
+export function generatePassportMiddlewares({keycloakOpts, localUsers}) {
+  const logger = createLogger();
+
+
+  if (keycloakOpts && typeof keycloakOpts === 'object' && validateKeycloakOpts(keycloakOpts)) {
+    return initKeycloakMiddleware(keycloakOpts);
+  }
+
+  if (typeof localUsers === 'string') {
+    return initLocalMiddlewares(localUsers);
+  }
+
+  throw new Error('No configuration for passport strategies');
+
+  function validateKeycloakOpts(keycloakOpts) {
+    const {publicKey, algorithms, audience, issuer} = keycloakOpts;
+
+    if (!publicKey) {
+      return false;
+    }
+
+    if (!algorithms || !Array.isArray(algorithms) || algorithms.length === 0) {
+      return false;
+    }
+
+    if (!audience) {
+      return false;
+    }
+
+    if (!issuer) {
```

Comment:
Refactor 2 ifs first checks if any of publicKey, algorithms, audience or issuer is missing and returns false.
2nd checks algorithms array and length.